src/pools/D3MTrueFiV1Pool.t.sol

```diff
@@ -48,13 +50,92 @@ contract D3MTrueFiV1PoolTest is AddressRegistry, D3MPoolBaseTest {
         _setUpTrueFiDaiPortfolio();
 
         d3mTestPool = address(new D3MTrueFiV1Pool(address(dai), address(portfolio), hub));
+        // set address of d3mTestPool to true in whitelist mapping in global whitelist lender verifier
+        hevm.store(GLOBAL_WHITELIST_LENDER_VERIFIER, keccak256(abi.encode(d3mTestPool, 2)), bytes32(uint256(1)));
+        _mintTokens(DAI, address(d3mTestPool), 5 ether);
+    }
+
+    function test_deposit_transfers_funds() public {
+        uint256 fundsBefore = dai.balanceOf(d3mTestPool);
+        D3MTrueFiV1Pool(d3mTestPool).deposit(1 ether);
+        uint256 fundsAfter = dai.balanceOf(d3mTestPool);
+
+        assertEq(portfolio.value(), 1 ether);
+        assertEq(fundsAfter, fundsBefore - 1 ether);
+    }
+
+    function test_deposit_issues_shares() public {
+        D3MTrueFiV1Pool(d3mTestPool).deposit(1 ether);
+        assertEq(uint256(ERC20Like(portfolio).balanceOf(d3mTestPool)), 1 ether);
+    }
+
+    function testFail_deposit_requires_auth() public {
+        D3MTrueFiV1Pool(d3mTestPool).deny(address(this));
+
+        D3MTrueFiV1Pool(d3mTestPool).deposit(1 ether);
+    }
+
+    function test_max_deposit_equals_max_size() public {
+        assertEq(D3MTrueFiV1Pool(d3mTestPool).maxDeposit(), portfolio.maxSize());
+    }
+
+    function test_max_desposit_equals_value_minus_deposited_funds() public {
+        D3MTrueFiV1Pool(d3mTestPool).deposit(1 ether);
+        assertEq(D3MTrueFiV1Pool(d3mTestPool).maxDeposit(), portfolio.maxSize() - 1 ether);
+    }
+
+    function test_withdraw_returns_funds() public {
+        D3MTrueFiV1Pool(d3mTestPool).deposit(1 ether);
+
+        uint256 fundsBefore = dai.balanceOf(d3mTestPool);
+        hevm.warp(block.timestamp + 30 days + 1 days);
+        D3MTrueFiV1Pool(d3mTestPool).withdraw(1 ether);
+        uint256 fundsAfter = dai.balanceOf(d3mTestPool);
+
+        assertEq(fundsAfter, fundsBefore + 1 ether);
+    }
+
+    function test_withdraw_burns_shares() public {
+        D3MTrueFiV1Pool(d3mTestPool).deposit(1 ether);
+
+        hevm.warp(block.timestamp + 30 days + 1 days);
```

Comment:
do we really need to do a warp here ?

Reply:
A depositor can only withdraw once the portfolio is closed, and portfolio only closes when the endDate is passed.